lib/active_merchant/billing/gateways/card_connect.rb

```diff
@@ -0,0 +1,283 @@
+
+      def verify(credit_card, options = {})
+        authorize(0, credit_card, options)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript
+          .gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]')
+          .gsub(%r(("cvv2\\":\\")\d*(&?)), '\1[FILTERED]')
+          .gsub(%r(("merchid\\":\\")\d*(&?)), '\1[FILTERED]')
+          .gsub(%r((&?"account\\":\\")\d*(&?)), '\1[FILTERED]')
+          .gsub(%r((&?"token\\":\\")\d*(&?)), '\1[FILTERED]')
+      end
+
+      private
+
+      def add_customer_data(post, options)
+        post[:email] = options[:email] if options[:email]
+      end
+
+      def add_address(post, options)
+        address = options[:billing_address] || options[:address]
+        if address
+          post[:address] = address[:address1] if address[:address1]
+          post[:city] = address[:city] if address[:city]
+          post[:region] = address[:state] if address[:state]
+          post[:country] = address[:country] if address[:country]
+          post[:postal] = address[:zip] if address[:zip]
+          post[:phone] = address[:phone] if address[:phone]
+        end
+      end
+
+      def add_money(post, money)
+        post[:amount] = amount(money)
+      end
+
+      def add_currency(post, money, options)
+        post[:currency] = (options[:currency] || currency(money))
+      end
+
+      def add_invoice(post, options)
+        post[:orderid] = options[:order_id]
+        post[:ecomind] = (options[:recurring] ? 'R' : 'E')
+      end
+
+      def add_payment(post, payment)
+        post[:name] = payment.name
+        if card_brand(payment) == 'check'
+          add_echeck(post, payment)
+        else
+          post[:account] = payment.number
+          post[:expiry] = expdate(payment)
+          post[:cvv2] = payment.verification_value
+        end
+      end
+
+      def add_echeck(post, payment)
+        post[:accttype] = 'ECHK'
+        post[:account] = payment.account_number
+        post[:bankaba] = payment.routing_number
+      end
+
+      def add_reference(post, authorization)
+        post[:retref] = authorization
+      end
+
+      def add_additional_data(post, options)
+        post[:ponumber] = options[:po_number]
+        post[:taxamnt] = options[:tax_amount] if options[:tax_amount]
+        post[:frtamnt] = options[:freight_amount] if options[:freight_amount]
+        post[:dutyamnt] = options[:duty_amount] if options[:duty_amount]
+        post[:orderdate] = options[:order_date] if options[:order_date]
+        post[:shipfromzip] = options[:ship_from_zip] if options[:ship_from_zip]
+        if (shipping_address = options[:shipping_address])
+          post[:shiptozip] = shipping_address[:zip]
+          post[:shiptocountry] = shipping_address[:country]
+        end
+        if options[:items]
+          post[:items] = []
+          options[:items].each do |item|
+            updated = {}
+            item.each_pair do |k, v|
```

Comment:
You likewise _can_ do this as a mapping operation, but playing with it locally, it was a bit gross. Best I got for the whole thing was

```ruby
post[:items] = options[:items].map do | item|
  item.map { |k, v| [k.to_s.gsub(/_/, ''), v] }.to_h
end
```

which wastes memory and I'm not convinced is cleaner, but your call. (There's a grosser version using `.inject` which doesn't waste RAM, but it gets sufficiently hard to read that I think your existing version is superior to that one.)